src/tables.jl

```diff
@@ -1,10 +1,24 @@
 using Tables: Tables
 
-Tables.istable(::Type{<:StructVector}) = true
-Tables.rowaccess(::Type{<:StructVector}) = true
-Tables.columnaccess(::Type{<:StructVector}) = true
+Tables.isrowtable(::Type{<:StructVector}) = true
 
-Tables.rows(s::StructVector) = s
+Tables.columnaccess(::Type{<:StructVector}) = true
 Tables.columns(s::StructVector) = fieldarrays(s)
 
 Tables.schema(s::StructVector) = Tables.Schema(staticschema(eltype(s)))
+
+function Base.append!(s::StructVector, rows)
+    if Tables.isrowtable(rows) && Tables.columnaccess(rows)
```

Comment:
Do we need to check `Tables.isrowtable` rather than just `Tables.istable`? Is it a matter of making sure that we are not adding functionality, but that the fallback of `append!(v::AbstractVector, itr)` would have worked and given the same result?

Reply:
This depends on your preference.  In particular, what do you want to happen with `append!(array, (a=[1], b=[2]))`?  Current implementation tries to reflect what standard array does: i.e., try to interpret the second argument as an iterator:

```julia
julia> append!(Any[], (a=[1], b=[2]))
2-element Array{Any,1}:
 [1]
 [2]

julia> append!(Int[], (a=[1], b=[2]))
ERROR: MethodError: Cannot `convert` an object of type Array{Int64,1} to an object of type Int64
```

_If_ we use `Tables.istable` here, we'd have

```julia
julia> append!(StructArray(a=Int[], b=Int[]), (a=[1], b=[2]))
1-element StructArray(::Array{Int64,1}, ::Array{Int64,1}) with eltype NamedTuple{(:a, :b),Tuple{Int64,Int64}}:
 (a = 1, b = 2)
```

This is a bit strange if you think `StructArray` is an optimized version of `Array` for structs.

Note that with this PR `append!(StructArray(a=Int[], b=Int[]), Tables.rows((a=[1], b=[2])))` still works (once https://github.com/JuliaData/Tables.jl/pull/153 is merged).

Reply:
You are right, a table could even iterate something other than rows, in which case we would just be doing something wrong here, it's safer to test `isrowtable`.